packages/react/src/components/toggle.tsx

```diff
@@ -0,0 +1,203 @@
+import React from "react";
+import { cva, type VariantProps } from "class-variance-authority";
+import { clx } from "../utils";
+
+const toggleVariants = cva(
+  "relative inline-flex items-center rounded-full focus:outline-none focus:ring-[3px]",
+  {
+    variants: {
+      size: {
+        medium: "w-[30px] h-[20px]",
+        large: "w-9 h-6",
+      },
+      checked: {
+        true: "bg-bg-primary-600 hover:bg-primary-700",
+        false: "bg-otl-gray-200 hover:bg-otl-gray-300",
+      },
+      disabled: {
+        true: "cursor-not-allowed",
+        false: "cursor-pointer",
+      },
+    },
+    compoundVariants: [
+      {
+        checked: true,
+        disabled: true,
+        className: "bg-bg-primary-disabled",
+      },
+      {
+        checked: false,
+        disabled: true,
+        className: "bg-bg-washed",
+      },
+    ],
+    defaultVariants: {
+      size: "medium",
+      checked: false,
+      disabled: false,
+    },
+  },
+);
+
+const thumbVariants = cva(
+  "absolute left-[3px] top-[3px] z-10 rounded-full transition-transform duration-200 ease-in-out",
+  {
+    variants: {
+      size: {
+        medium: "w-[14px] h-[14px]",
+        large: "w-[18px] h-[18px]",
+      },
+      checked: {
+        true: "",
+        false: "",
+      },
+      disabled: {
+        true: "bg-bg-white",
+        false: "bg-white",
+      },
+    },
+    compoundVariants: [
+      {
+        size: "medium",
+        checked: true,
+        className: "translate-x-[10px]",
+      },
+      {
+        size: "large",
+        checked: true,
+        className: "translate-x-3",
+      },
+    ],
```

Comment:
use `data-[state]` to set & listen for the change in `checked` state

Reply:
done boss